scripts/check_unreferenced_files.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+SCRIPT_DIR=$(dirname ${0})
+BASE_DIR=${SCRIPT_DIR}/..
+
+unreferenced=0
+
+function get_filenames {
+  find $BASE_DIR/$1 -maxdepth 1 -type f -name "*.$2" -or -name "*.$2.in" | xargs -IFILE git ls-files FILE | xargs -IFILE basename FILE
+}
+
+function check_directory {
+  # return value from grep --files-without-match seems to differ
+  # between grep versions so we use output instead of return value
+  matches=$(get_filenames $1 $2 | xargs -IFILE sh -c "grep --files-without-match FILE $BASE_DIR/$1/CMakeLists.txt | sed -e 's!^.*/CMakeLists.txt\$!FILE!'")
```

Comment:
This is very hard to follow, but I have no immediate suggestion for an alternative. Maybe use a simple for-loop and just check for the occurrence of each file in the `CMakeLists.txt` file?